src/java/org/apache/cassandra/audit/AuditLogOptionsCompositeData.java

```diff
@@ -0,0 +1,162 @@
+import javax.management.openmbean.SimpleType;
+
+import org.apache.cassandra.config.ParameterizedClass;
+
+public class AuditLogOptionsCompositeData
+{
+    private static final String[] ITEM_NAMES = new String[]{ "audit_logs_dir",
+                                                             "archive_command",
+                                                             "roll_cycle",
+                                                             "block",
+                                                             "max_queue_weight",
+                                                             "max_log_size",
+                                                             "max_archive_retries",
+                                                             "enabled",
+                                                             "included_keyspaces",
+                                                             "excluded_keyspaces",
+                                                             "included_categories",
+                                                             "excluded_categories",
+                                                             "included_users",
+                                                             "excluded_users",
+                                                             "logger" };
+
+    private static final String[] ITEM_DESC = new String[]{ "directory where audit data are stored",
+                                                            "archive command for audit data",
+                                                            "how often to roll BinLog segments so they can potentially be reclaimed",
+                                                            "indicates if the BinLog should block if the it falls behind or should drop bin log records",
+                                                            "maximum weight of in memory queue for records waiting to be written to the binlog file before blocking or dropping the log records",
+                                                            "maximum size of the rolled files to retain on disk before deleting the oldest file",
+                                                            "number of times to retry an archive command",
+                                                            "boolean telling if we are enabled or not",
+                                                            "included keyspaces",
+                                                            "excluded keyspaces",
+                                                            "included categories",
+                                                            "excluded categories",
+                                                            "included users",
+                                                            "excluded users",
+                                                            "audit logger implementation class name" };
```

Comment:
Rather than working with three arrays, could we introduce a class like
```
class AuditLogOption
{
    final String name;
    final String description;
    final SimpleType type;
}
```

Reply:
I do not want to do this differently, actually. This is done like it is here across the whole project, I just do not want to introduce just another way how we do it, it is just good enough how it is, in my opinion.

Reply:
The suggestion is trivial, and was offered based on providing a) improved readability, and b) safety (if items are added).

(a) certainly weighs against what the precedence in the codebase already is. Might you provide examples of the precedence that is "across the whole project", as the use of multiple separate arrays being used consistently instead of simple structs surprises me a little?

Reply:
I ll find some cases like that for you on Monday.